tests/lib/c_lib/src/test_qsort.c

```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define _GNU_SOURCE
```

Comment:
This test only tests the minimal libc, so this should not be necessary.

https://github.com/zephyrproject-rtos/zephyr/blob/cd610cb5ae148b8634c130205908ffad30bf1d1e/tests/lib/c_lib/testcase.yaml#L2-L4

Reply:
Ok, I can move this change into the patch adding picolibc to the testcast.yaml file. However, if the minimal libc is ever changed to conform to the POSIX compilation environment requirements, some feature test macro would be necessary or the qsort_r function would not be visible. Happy to move it or leave it as you prefer.